Content.Server/EntityEffects/Effects/FlammableReaction.cs

```diff
@@ -28,7 +28,7 @@ public override void Effect(EntityEffectBaseArgs args)
             if (!args.EntityManager.TryGetComponent(args.TargetEntity, out FlammableComponent? flammable))
                 return;
 
-            var multiplier = flammable.FireStacks == 0f ? Multiplier : MultiplierOnExisting;
+            var multiplier = flammable.FireStacks != 0f && MultiplierOnExisting >= 0 ? MultiplierOnExisting : Multiplier;
             var quantity = 1f;
```

Comment:
Please document magic number changes like this because it makes discerning intent a nightmare. Same with the datafield.

Reply:
Where would you like the numbers documented? In a comment or some other place?

Reply:
One here explaining what's supposed to happen and another on the field would be good.